assets/js/components/notifications/BannerNotifications.test.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * `BannerNotifications` tests.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	createTestRegistry,
+	muteFetch,
+	provideModules,
+	provideSiteConnection,
+	provideSiteInfo,
+	provideUserAuthentication,
+	provideUserCapabilities,
+	provideUserInfo,
+	render,
+} from '../../../../tests/js/test-utils';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import useQueryArg from '../../hooks/useQueryArg';
+import BannerNotifications from './BannerNotifications';
+
+jest.mock( '../../hooks/useQueryArg' );
+
+// Set up mockImplementation for `useQueryArg` used in this component,
+// so we can set expected the query params
+function stubMockUseQueryArg( notification = '' ) {
```

Comment:
As a general rule we try to avoid stubbing/mocking as much as we can in our tests. With that in mind, I'd suggest taking an alternative approach here to mock the location object instead and avoid stubbing `useQueryArg()` - still an element of mocking, but at a lower level :)

So, I would suggest making use of the existing `mockLocation()` helper, and simply assigning to `global.location.href` for each test, along these lines:

```diff
diff --git a/assets/js/components/notifications/BannerNotifications.test.js b/assets/js/components/notifications/BannerNotifications.test.js
index 27eaaddedd..dfed41ee3d 100644
--- a/assets/js/components/notifications/BannerNotifications.test.js
+++ b/assets/js/components/notifications/BannerNotifications.test.js
@@ -19,6 +19,7 @@
 /**
  * Internal dependencies
  */
+import { mockLocation } from '../../../../tests/js/mock-browser-utils';
 import {
 	createTestRegistry,
 	muteFetch,
@@ -32,23 +33,11 @@ import {
 } from '../../../../tests/js/test-utils';
 import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
 import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
-import useQueryArg from '../../hooks/useQueryArg';
 import BannerNotifications from './BannerNotifications';
 
-jest.mock( '../../hooks/useQueryArg' );
-
-// Set up mockImplementation for `useQueryArg` used in this component,
-// so we can set expected the query params
-function stubMockUseQueryArg( notification = '' ) {
-	useQueryArg.mockImplementation( ( queryArg ) => {
-		if ( 'notification' === queryArg && notification ) {
-			return [ notification ];
-		}
-		return [ undefined ];
-	} );
-}
-
 describe( 'BannerNotifications', () => {
+	mockLocation();
+
 	let registry;
 
 	beforeEach( () => {
@@ -78,7 +67,8 @@ describe( 'BannerNotifications', () => {
 	} );
 
 	it( 'render `authentication_success` notification if `authentication_success` query param value is passed', async () => {
-		stubMockUseQueryArg( 'authentication_success' );
+		global.location.href =
+			'http://example.com/wp-admin/admin.php?notification=authentication_success';
 
 		const { queryByText, waitForRegistry } = render(
 			<BannerNotifications />,
@@ -97,7 +87,8 @@ describe( 'BannerNotifications', () => {
 	it( 'does not render `setup complete` notification if `custom_dimensions` query arg value is passed', async () => {
 		// Add arbitrary value for `notification` to prevent server appending `authentication_success`
 		// on the redirect, so setup completed notification does not show
-		stubMockUseQueryArg( 'custom_dimensions' );
+		global.location.href =
+			'http://example.com/wp-admin/admin.php?notification=custom_dimensions';
 
 		const { queryByText, waitForRegistry } = render(
 			<BannerNotifications />,
```

Reply:
@techanvil Ha, thanks! It didn't occur to me to set global location. I did originally use `mockLocation` but to implement `navigateTo` like in other tests, then when I noticed component does not have location set, had to change it, so I went by mocked version of `useQueryArgs`.  But your suggestion is much better :D